pkg/registry/controller/etcd/etcd.go

```diff
@@ -95,3 +119,74 @@ func (r *StatusREST) New() runtime.Object {
 func (r *StatusREST) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool, error) {
 	return r.store.Update(ctx, obj)
 }
+
+type ScaleREST struct {
+	registry *controller.Registry
+}
+
+// ScaleREST implements Patcher
+var _ = rest.Patcher(&ScaleREST{})
+
+// New creates a new Scale object
+func (r *ScaleREST) New() runtime.Object {
+	return &autoscaling.Scale{}
+}
+
+func (r *ScaleREST) Get(ctx api.Context, name string) (runtime.Object, error) {
+	rc, err := (*r.registry).GetController(ctx, name)
+	if err != nil {
+		return nil, errors.NewNotFound(autoscaling.Resource("replicationcontrollers/scale"), name)
+	}
+	return &autoscaling.Scale{
+		ObjectMeta: api.ObjectMeta{
+			Name:              name,
+			Namespace:         rc.Namespace,
+			CreationTimestamp: rc.CreationTimestamp,
+		},
+		Spec: autoscaling.ScaleSpec{
+			Replicas: rc.Spec.Replicas,
+		},
+		Status: autoscaling.ScaleStatus{
+			Replicas: rc.Status.Replicas,
+			Selector: labels.SelectorFromSet(rc.Spec.Selector).String(),
+		},
+	}, nil
+}
+
+func (r *ScaleREST) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool, error) {
+	if obj == nil {
+		return nil, false, errors.NewBadRequest("nil update passed to Scale")
+	}
+	scale, ok := obj.(*autoscaling.Scale)
+	if !ok {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("wrong object passed to Scale update: %v", obj))
+	}
+
+	if errs := validation.ValidateScale(scale); len(errs) > 0 {
+		return nil, false, errors.NewInvalid(autoscaling.Kind("ScaleTwo"), scale.Name, errs)
+	}
+
+	rc, err := (*r.registry).GetController(ctx, scale.Name)
+	if err != nil {
+		return nil, false, errors.NewNotFound(autoscaling.Resource("replicationcontrollers/scale"), scale.Name)
+	}
+	rc.Spec.Replicas = scale.Spec.Replicas
+	rc, err = (*r.registry).UpdateController(ctx, rc)
+	if err != nil {
+		return nil, false, errors.NewConflict(autoscaling.Resource("replicationcontrollers/scale"), scale.Name, err)
+	}
+	return &autoscaling.Scale{
+		ObjectMeta: api.ObjectMeta{
+			Name:              rc.Name,
```

Comment:
> If there's a good reason to not make this metadata consistent with autoscaling/v1 Scale, please add a comment.

I have made some changes. I still need some clarification about this comment. I will talk to you about it.

> I think the main thing to check is whether update conflicts could confuse the 1.1 HPA controller.

The HPA code as it is right now can potentially confuse 1.1 HPA controller. This is because the Get and Update to Scale in HPA 1.1 completely ignored group/version, so the RC object here and in 1.1 HPA can point to the same underlying etcd object. I am not sure how to fix this problem though.

> Also, this looks like it should be a helper since it's the same as Get.

Pulled Scale object instantiation into a helper.